compute/cloud-client/src/test/java/compute/snapshotschedule/SnapshotScheduleIT.java

```diff
@@ -0,0 +1,174 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.compute.v1.DeleteResourcePolicyRequest;
+import com.google.cloud.compute.v1.GetResourcePolicyRequest;
+import com.google.cloud.compute.v1.InsertResourcePolicyRequest;
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.Operation.Status;
+import com.google.cloud.compute.v1.ResourcePoliciesClient;
+import com.google.cloud.compute.v1.ResourcePolicy;
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.MockedStatic;
+
+@RunWith(JUnit4.class)
+@Timeout(value = 2, unit = TimeUnit.MINUTES)
+public class SnapshotScheduleIT {
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String REGION = "us-central1";
+  private static final String SCHEDULE_NAME = "test-schedule-" + UUID.randomUUID();
+  private static final String SCHEDULE_DESCRIPTION = "Test hourly snapshot schedule";
+  private static final int MAX_RETENTION_DAYS = 2;
+  private static final String STORAGE_LOCATION = "US";
+  private static final String ON_SOURCE_DISK_DELETE = "KEEP_AUTO_SNAPSHOTS";
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+            .that(System.getenv(envVarName)).isNotEmpty();
+  }
+
+  @BeforeAll
+  public static void setUp()
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+    CreateSnapshotSchedule.createSnapshotSchedule(
+            PROJECT_ID, REGION, SCHEDULE_NAME, SCHEDULE_DESCRIPTION,
+            MAX_RETENTION_DAYS, STORAGE_LOCATION, ON_SOURCE_DISK_DELETE);
+  }
+
+  @AfterAll
+  public static void cleanup()
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    DeleteSnapshotSchedule.deleteSnapshotSchedule(PROJECT_ID, REGION, SCHEDULE_NAME);
+  }
+
+  @Test
+  public void testEditSnapshotSchedule()
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    Status status = EditSnapshotSchedule.editSnapshotSchedule(
+            PROJECT_ID, REGION, SCHEDULE_NAME);
+
+    assertThat(status).isEqualTo(Status.DONE);
```

Comment:
Suggesting to replace checking the status with another assertion. Status may not be always Done and this will cause flakiness.

Reply:
Thank you for comment. Added new assertions

Reply:
Kindly remove the assertion for Status.

Reply:
Deleted